api/swagger.yaml

```diff
@@ -8897,6 +8897,34 @@ paths:
           required: true
           type: "string"
       tags: ["Exec"]
+  /exec/{id}/kill:
+    post:
+      summary: "Kill an exec instance"
```

Comment:
The naming and graceful shutdown issues you mentioned above seem to be related, so I'll respond to both here.

As you noted above, `/container/{id}/kill` has some extra handling to wait for the container to stop if a `SIGKILL` signal is used. It is somewhat surprising behavior to me that which signal is sent changes other behavior (and this doesn't seem to be noted in the endpoint's documentation). The endpoint does feel a bit muddled to me - is it intended to behave like the Linux `kill` command, or is it intended to kill the process? It can do either one, but some behaviors, such as "send SIGKILL and don't wait" don't seem to be possible through the API. All this seems to stem from the confusion regarding the name.

Given that, this implementation of exec kill might indeed be better described as `/exec/{id}/signal` in order to help differentiate it. A second endpoint, `/exec/{id}/stop` might also be useful, mirroring the behavior of the equivalent container API for graceful shutdown.

Reply:
Taking a further look, some more of the container APIs would probably make sense to have `exec` equivalents:

* `/exec/{id}/attach`
* `/exec/{id}/wait`

I don't know that we want to hold up this PR to create all these, though, as I expect the `signal` endpoint, combined with polling the existing inspect endpoint if the exit status is needed, will fulfill a lot of users needs.